api/firmware/eth.go

```diff
@@ -78,24 +79,72 @@ func (device *Device) ETHSign(
 	recipient [20]byte,
 	value *big.Int,
 	data []byte) ([]byte, error) {
+	supportsAntiklepto := device.version.AtLeast(semver.NewSemVer(9, 5, 0))
```

Comment:
This condition for the version at least 9.5.0 is in multiple places. Consider making a device method like `isAntikleptoSupported() bool`, maybe even an exported method for others to use.

Reply:
It is specific to these api endpoints though. In BTC tx signing for example, it is supported from 9.4.0.

> maybe even an exported method for others to use.

There will not be other uses, 9.5.0 adds support for exactly these three endpoints.

Still think I should move it to a function?

Reply:
I guess I just don't really like all these semver's around:

![image](https://user-images.githubusercontent.com/25405/108405819-db14e900-7221-11eb-9222-79724235a2e2.png)

Would consider maybe a list of globals, something like this:

```go
type supportedFeature int

const (
  featureAttestation        // device attestation
  featureNeedsOpNoise       // device requires OP noise byte
  featureBTCAntikleptoMsg   // anti-klepto support for msg signing
  featureBTCAntikleptoTx    // anti-klepto support for tx signing
  ... and so on
)

// featureSemVer defines semantic versions which every feature was added at. 
// Use Device.IsSupported to identify whether a feature is supported by a device. 
var featureSemVer map[supportedFeature]semver.SemVer{
  featureAttestation: semver.NewSemVer(2, 0, 0),
  featureNeedsOpNoise: semver.NewSemVer(4, 0, 0),
  featureBTCAntikleptoMsg: semver.NewSemVer(9, 4, 0),
  featureBTCAntikleptoTx: semver.NewSemVer(9, 5, 0),
  ...
)

// IsSupported reports whether the feature t is supported by the device d.
func (d *Device) IsSupported(t supportedFeature) bool {
  v, ok := featureSemVer[t]
  return ok && d.version.AtLeast(v)
}

// elsewhere:
if device.IsSupported(featureBTCAntikleptoTx) {
 ...
}
```

I'd find it more pleasant to read and it's very good to have a clear list of when each feature was added imho. Might come in handy later or just useful for those like me who don't know the whole history.

This obviously doesn't fit in the PR.